dms/tests/common.py

```diff
@@ -227,11 +227,14 @@ def create_storage(self, save_type="database"):
             {"name": "Test Storage", "save_type": save_type}
         )
 
-    def create_directory(self, storage=False, directory=False, res_model=False):
+    def create_directory(self, storage=False, directory=False, model_id=False):
         record = Form(self.directory_model)
         record.name = uuid.uuid4().hex
         record.is_root_directory = True
-        record.res_model = res_model
```

Comment:
Keep this assignment to confirm that `model_id` is defined correctly and there are no permissions issues.

Reply:
Please check https://github.com/len-foss/dms/commit/fefc351c0e186fd5f6aab7c16203fc8aad780c98
The commit message mentioned the change in Form that ensures that this code snippet, which is wrong, cannot work in v16 and upwards.

Reply:
I agree with @len-foss. With his solution, the tests could go through. Here are the arguments for his solution: 

* in the form view of the directory: you can only set the **model_id**, not the **res_model**, because **res_model** is invisible. We get this error message if we keep the line. 

```bash

  File "/home/vancouver/*/*/*/*dms/tests/test_portal.py", line 13, in setUp
    super().setUp()
  File "/opt/odoo/lib/odoo/odoo/tests/common.py", line 1605, in setUp
    super().setUp()
  File "/home/vancouver/*/*/*/*dms/tests/common.py", line 277, in setUp
    self.create_directory(storage=self.storage, res_model="res.partner")
  File "/home/vancouver/*/*/*/*dms/tests/common.py", line 237, in create_directory
    record.res_model = res_model
    ^^^^^^^^^^^^^^^^
  File "/opt/odoo/lib/odoo/odoo/tests/common.py", line 2254, in __setattr__
    assert not self._get_modifier(field, 'invisible'), \
AssertionError: can't write on invisible field res_model
```

* Either we make **res_model** field editable in form view or keep those changes for the test